neo4j/transaction_with_context.go

```diff
@@ -70,26 +88,26 @@ func (tx *explicitTransaction) Run(ctx context.Context, cypher string, params ma
 	}
 	stream, err := tx.conn.RunTx(ctx, tx.txHandle, db.Command{Cypher: cypher, Params: params, FetchSize: tx.fetchSize})
 	if err != nil {
-		tx.err = err
-		tx.runFailed = true
-		tx.onClosed(tx)
-		return nil, errorutil.WrapError(tx.err)
+		tx.txState.onError(err)
+		tx.onClosed()
```

Comment:
```suggestion
```
Do we still need to call this, considering that session will register onClose to be called on `txState.onError`? It my assumption true for managed transactions as well?

Reply:
No, we can remove it, it gets closed when the `onError` is called 👍